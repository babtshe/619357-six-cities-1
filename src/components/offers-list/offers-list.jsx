import React from 'react';
import {OfferCard} from '../offer-card/offer-card';
import {propTypes} from './offers-list.props';


const OffersList = (props) => {
  const {
    offers,
    setActiveItem,
    unsetActiveItem,
    setClickedItem,
  } = props;

  return (
    <div className="cities__places-list places__list tabs__content">
      {offers.map((card) => {
        return <OfferCard
          onMouseEnter = {setActiveItem}
          onMouseLeave = {unsetActiveItem}
          key = {card.name}
          card = {card}
          onClick = {setClickedItem}
        />;
      })}
    </div>
  );
};

OffersList.propTypes = propTypes;

export default OffersList;
