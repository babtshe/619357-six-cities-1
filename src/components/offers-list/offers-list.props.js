import PropTypes from 'prop-types';
import {cardTypes} from '../offer-card/offer-card.props';

export const propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape(cardTypes)),
  setClickedItem: PropTypes.func,
  ClassPrefix: PropTypes.string,
};
