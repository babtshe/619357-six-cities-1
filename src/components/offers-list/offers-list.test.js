import React from 'react';
import renderer from 'react-test-renderer';
import {OffersList} from './offers-list';

it(`OffersList should render correctly`, ()=>{
  const mock = [
    {
      name: `test`,
      type: `Apartment`,
      price: 1,
      rating: 1,
      image: `test`,
      link: `#`,
      mark: `Premium`,
    },
    {
      name: `test`,
      type: `Private room`,
      price: 1,
      rating: 1,
      image: `test`,
      link: `#`,
      bookmarked: true,
    }
  ];
  const tree = renderer
  .create(
      <OffersList
        cards = {mock}
      />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
