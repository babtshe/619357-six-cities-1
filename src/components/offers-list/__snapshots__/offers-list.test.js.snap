// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersList should render correctly 1`] = `
<div
  className="cities__places-list places__list tabs__content"
/>
`;
